### DISCUSSION TITLE
How to map null to empty Array/List?

### DISCUSSION THREAD

**Discussion by viblo-majority** [riok/mapperly#1251] [Q&A] [answered]
Im porting over some Automapper mappings. AutoMapper will by default map a null value to empty container and not to a null value. Can I do this with Mapperly as well, and if so how?

E.g. this AutoMapper code will print `True, False`

```c#
using System;
using System.Collections.Generic;

public class A { public IEnumerable<string> X { get; set; } }
public class B { public IEnumerable<string> X { get; set; } }

public class Program
{	
	public static void Main()
	{
		var config = new AutoMapper.MapperConfiguration(
			cfg => {cfg.CreateMap<A, B>();});
				
		var mapper = config.CreateMapper();
		var a = new A();
		var b = mapper.Map<B>(new A());
		Console.WriteLine($"{a.X==null}, {b.X==null}");
	}
}
```

**Comment by latonz** [comment, marked as answer]
You can use a user-implemented mapping method for this, see [docs](https://mapperly.riok.app/docs/configuration/user-implemented-methods/). Unfortunately, generic user-implemented mappings are not yet supported in Mapperly, see https://github.com/riok/mapperly/issues/451.
